lib/utils/dom.js

```diff
@@ -2,6 +2,184 @@ var config = require('../config');
+};
+
+/**
+ * Gets the elements(Ids) which are controlled by elId.
+ * If 'elControlls' is null then there is nothing controlling the element passed in and so it is returned.
+ *
+ * @param      {HTMLElementId} elId Id of the element which controls other elements.
+ * @returns    {Array} { array of Ids }
+ */
+
+utils.getIdsOfControlledContainers = function(elId){
+	var elControls = document.getElementById(elId).getAttribute("aria-controls");
+	if(!elControls)
+		return null;
+	return elControls.split(" ");
+};
+
+/**
+ * Getst the element controlling the element passed in. If the controlling element is
+ *not accessible find the controlling element.
+ * @param      {HTMLElement} element you are trying to find the .
+ * @returns    {HTMLElementId} { Id }
+ */
+
+//passing in main-menu, splits out the same thing... why?
+utils.getControllingElementId = function(element){
+	if(!element.hasAttribute("aria-labelledby"))
+		return null;
+    var labelledBy = element.getAttribute("aria-labelledby");
+    var controllingEl = document.getElementById(labelledBy);
+
+    //If controlling El is disabled/hidden then find its parent (controlling) element
+	if(!isNavigable(controllingEl)){
+			var nextControllingEl = utils.closestWithClass(controllingEl, "menu");
+			if(nextControllingEl)
+				return utils.getControllingElementId(nextControllingEl);
+	}
+
+	return labelledBy;
+};
+
+/**
+* If the element passed in is accessible then true is retruned.
+ * @param      {HTMLElement} element
+ * @returns    {boolean} { boolean }
+ */
+function isNavigable(element){
+	if(element.hasAttribute("aria-disabled")){
+		if(element.getAttribute("aria-disabled") === "true")
+			return false;
+	}
+	if(element.hasAttribute("aria-hidden")){
+		if(element.getAttribute("aria-hidden") === "true")
+			return false;
+	}
+	return true;
+}
+/**
+* If `index` is the first item within links, then the last items index within links will be returned.
+* If `index` is the last item within links, then the first items index within links will be returned.
+ * @param      {int} current index
+ * @param      {links} list of elements
+ * @returns    {int} { index }
+ */
+function wrapToEnd(index, links){
+		if(index > links.length - 1)
+			index = 0;
+		if(index < 0)
+			index = links.length - 1;
+		return index;
+}
+
+/**
+*gets the index of the next accessible element within a list of elements in relation to direction
+ * @param      {int}int  current index of element
+ * @param      {Array} list of elements
+ * @param      {int} direction which the index is trying to move -1 being the previous index
+ * 1 being the nect index
+ * @returns    {int} { index }
+ */
+function skipDisabledLinks(index,links, skipDirectionInt){
+	while(!isNavigable(links[index])){
+		index += skipDirectionInt;
+		index = wrapToEnd(index, links);
+	}
+	return index;
+}
+
+/**
+ * Gets the links from within the container, as an Array.
+ *
+ * @param      {HTMLElement}  container    The containing element
```

Comment:
This comment needs updating to match the methos - seems to be a list of containers not a single container